test/hotspot/jtreg/compiler/loopopts/superword/TestEquivalentInvariants.java

```diff
@@ -735,15 +802,34 @@ static Object[] testMemorySegmentIInvarL3d3(MemorySegment m, int size) {
         return new Object[]{ m };
     }
 
+    @Test
+    @IR(counts = {IRNode.LOAD_VECTOR_I, "= 10",
+                  IRNode.STORE_VECTOR,  "= 10"},
+        applyIf = {"AlignVector", "false"},
+        applyIfPlatform = {"64-bit", "true"},
+        applyIfCPUFeatureOr = {"sse4.1", "true", "asimd", "true"})
+    static Object[] testMemorySegmentIInvarL3e(MemorySegment m, int invar1, int invar2, int invar3, int size) {
+        long i1 = (long)(-invar1 + invar2 + invar3);
+        long i2 = (long)(invar2 + invar3) - (long)(invar1); // not equivalent
+        for (int i = 0; i < size; i+=2) {
+            int v0 = m.getAtIndex(ValueLayout.JAVA_INT, i + i1 + 0);
+            int v1 = m.getAtIndex(ValueLayout.JAVA_INT, i + i2 + 1);
+            m.setAtIndex(ValueLayout.JAVA_INT, i + i1 + 0, v0 + 1);
+            m.setAtIndex(ValueLayout.JAVA_INT, i + i2 + 1, v1 + 1);
+        }
+        return new Object[]{ m };
+    }
+
     @Test
     @IR(counts = {IRNode.LOAD_VECTOR_I, "= 0",
                   IRNode.STORE_VECTOR,  "= 0"},
+        applyIf = {"AlignVector", "true"},
         applyIfPlatform = {"64-bit", "true"},
         applyIfCPUFeatureOr = {"sse4.1", "true", "asimd", "true"})
     // Should never vectorize, since i1 and i2 are not guaranteed to be adjacent
     // invar2 + invar3 could overflow, and the address be valid with and without overflow.
     // So both addresses are valid, and not adjacent.
```

Comment:
This needs extra attention: it _does_ vectorize with `applyIf = {"AlignVector", "false"}`, so something is off.